third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1340,501 +1341,1256 @@ pub struct CodeUnit {
 ///
 /// Bytecodes operate on a stack machine and each bytecode has side effect on the stack and the
 /// instruction stream.
+#[bytecode_spec]
 #[derive(Clone, Hash, Eq, VariantCount, PartialEq)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
 #[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
 pub enum Bytecode {
-    /// Pop and discard the value at the top of the stack.
-    /// The value on the stack must be an copyable type.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., value -> ...```
+    #[group = "stack_and_local"]
+    #[description = "Pop and discard the value at the top of the stack. The value on the stack must be an copyable type."]
+    #[semantics = "stack >> _"]
+    #[paranoid_post = r#"
```

Comment:
I'd prefer not calling this `paranoid`. Maybe `runtime_type_check_epilogue` and `runtime_type_check_prologue`. Calling it just `paranoid` (not even `paranoid_type_check`).